atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java

```diff
@@ -591,6 +593,67 @@ public void testSweepTimestampMetric() {
         TargetedSweepMetricsTest.assertSweepTimestampThoroughEquals(5);
     }
 
+    @Test
+    public void doNotSweepAnythingAfterEntryWithCommitTsAfterSweepTs() {
+        immutableTs = 1000L;
+        enqueueWrite(TABLE_CONS, 900);
+        enqueueWrite(TABLE_CONS, 910);
+        enqueueWriteCommitedAt(TABLE_CONS, 950, 1000);
+        enqueueWrite(TABLE_CONS, 970);
+        enqueueWrite(TABLE_CONS, 920);
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertThat(progress.getLastSweptTimestamp(ShardAndStrategy.conservative(CONS_SHARD))).isEqualTo(920L);
+        ArgumentCaptor<Map> argument = ArgumentCaptor.forClass(Map.class);
+        verify(spiedKvs, times(1)).deleteAllTimestamps(eq(TABLE_CONS), argument.capture(), eq(false));
+        assertThat(argument.getValue()).containsValue(920L);
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertThat(progress.getLastSweptTimestamp(ShardAndStrategy.conservative(CONS_SHARD))).isEqualTo(920L);
+        verify(spiedKvs, times(1)).deleteAllTimestamps(any(TableReference.class), anyMap(), eq(false));
+
+        immutableTs = 1101L;
```

Comment:
Would 1001L make more sense to catch off by one errors and more precisely check the logic?

Reply:
Sure